### PR TITLE
[Element] unify getQuery() method on listing DAOs

### PR DESCRIPTION
**Pull request by brusch**

### CODE REVIEW

**Comment by weisswurstkanone** [comment]
please have a look at the tests

File: models/DataObject/Listing/Dao.php [hunk 1]
```diff
@@ -173,7 +173,7 @@ public function getCount()
      */
     public function loadIdList()
     {
-        $query = $this->getQuery();
+        $query = $this->getQuery([new Expression(sprintf('%s as o_id', $this->getTableName() . '.o_id')), 'o_type']);
```
Comment (weisswurstkanone) [inline models/DataObject/Listing/Dao.php]: This looks odd, array or string ?

image

File: models/DataObject/Listing/Concrete/Dao.php [hunk 2]
```diff
@@ -51,25 +51,20 @@ class Dao extends Model\DataObject\Listing\Dao
...
         $select = $this->db->select();
 
-        // create base
-        $field = $this->getTableName() . '.o_id';
         $select->from(
             [$this->getTableName()],
-            [
-                new Expression(sprintf('%s as o_id', $this->getSelectPart($field, $field))), 'o_type'
-            ]
+            [$columns]
```
Comment (weisswurstkanone) [inline models/DataObject/Listing/Concrete/Dao.php]: what if columns is already an array ?

File: models/DataObject/Listing/Dao.php (hunk 1 above)
Comment (brusch) [reply models/DataObject/Listing/Dao.php]: @weisswurstkanone well, that comes from here: https://github.com/pimcore/pimcore/blob/934c42b6cb2085cbd96889f542e4b888c4e02689/lib/Db/ZendCompatibility/QueryBuilder.php#L356

**Pull request merged**